### ISSUE TITLE
cockpit-tls man page is out of date, now that cockpit reads the certificate & private key from separate files

### ISSUE THREAD

**Issue by yrro** [cockpit-project/cockpit#17358]
### Explain what happens

Taking the example from `cockpit-tls(8)`:

```
# getcert request -c IPA -f /etc/pki/tls/certs/hostname.crt -F /etc/pki/tls/certs-chain.crt -k /etc/pki/tls/private/hostname.key -g 3072 -C 'sed -n w/etc/cockpit/ws-certs.d/50-from-certmonger.cert /etc/pki/tls/certs/hostname.crt /etc/pki/tls/certs/hostname-chain.crt /etc/pki/tls/private/hostnamekey'
```

Results in:

```
$ ls -l /etc/cockpit/ws-certs.d/
[...]
-rw-------. 1 root root        12334 May 19 17:22 50-from-certmonger.cert
```

Cockpit can't read the file because it's not created with the group `cockpit-ws` and it's not group-readable.

It seems since that man page was written, cockpit now has support for separate certificate & private key files. This worked for me:

```
[root@myhost ~]# getcert request -v -w -g 3072 -c IPA -k /etc/cockpit/ws-certs.d/51-myhost.key -o cockpit-wsinstance -f /etc/cockpit/ws-certs.d/51-myhost.crt -M 0644                                                                         
New signing request "20220519165212" added.
State NEWLY_ADDED_READING_KEYINFO, stuck: no.
State GENERATING_KEY_PAIR, stuck: no.
State GENERATING_CSR, stuck: no.
State SUBMITTING, stuck: no.
State READING_CERT, stuck: no.
State MONITORING, stuck: no.

[root@myhost ~]# /usr/libexec/cockpit-certificate-ensure --check 
Would use certificate /etc/cockpit/ws-certs.d/51-myhost.crt
```

I also had to restart `cockpit` in order for the certificate to start being used. SELinux policy doesn't allow certmonger to restart services, so that would have to be done with a separate script file, with the `certmonger_unconfined_exec_t` context in order for certificate rotation to be truly automated.

NB, it would be nice if `cockpit-certificate-ensure` output both the private key file as well as the certificate file.

### Version of Cockpit

264

### Where is the problem in Cockpit?

Documentation

### Server operating system

Red Hat Enterprise Linux

### Server operating system version

8.6

**Comment by martinpitt** [comment]
> Cockpit can't read the file because it's not created with the group cockpit-ws and it's not group-readable.

Did you actually see it fail, or was this just some assumption? This should not be an issue any more in general -- [version 257](https://cockpit-project.org/blog/cockpit-257.html) changed the handling of the certificates to not require any particular permissions any more, so that they can be shared between services.

Indeed the example from the manpage is outdated -- we don't recommend any more to install a merged cert/key, but split them (as you did). I'll use this issue for tracking the update of that.

Thanks for reporting!

**Comment by yrro** [comment]
> Did you actually see it fail, or was this just some assumption?

Yes, with version 264. But since this doesn't affect versions 257 and onwards, I think you can just adjust the sample `getcert` command to put the certificate and key files directly into `/etc/cockpit/ws-certs.d` and drop the post-save command.

> Indeed the example from the manpage is outdated -- we don't recommend any more to install a merged cert/key, but split them (as you did). I'll use this issue for tracking the update of that.

Makes sense, I'll retitle accordingly, thanks.

**Comment by martinpitt** [comment]
Fixed in #17369

**Comment by yrro** [comment]
> Yes, with version 264. But since this doesn't affect versions 257 and onwards, I think you can just adjust the sample `getcert` command to put the certificate and key files directly into `/etc/cockpit/ws-certs.d` and drop the post-save command.

Hold on, I seem to have forgotten how numbers work! I can't reproduce this by setting both certificate & key to be owned by root with mode 0600 & restarting cockpit. So never mind about that, if I ever figure out what the problem was I'll open a separate issue.
